rpc/api/eth.go

```diff
@@ -406,6 +407,56 @@ func (self *ethApi) Call(req *shared.Request) (interface{}, error) {
 	}
 }
 
+func (self *ethApi) DebugTransaction(req *shared.Request) (interface{}, error) {
+
+	args := new(HashArgs)
+	if err := self.codec.Decode(req.Params, &args); err != nil {
+		return nil, shared.NewDecodeParamError(err.Error())
+	}
+
+	vmtrace, ret, gas, err := self.xeth.DebugTransaction(args.Hash)
+
+	if err != nil {
+		return nil, err
+	}
+
+	res := TransactionExecutionRes{
+		Gas:         gas,
+		ReturnValue: fmt.Sprintf("%x", ret),
+		StructLogs:  make([]StructLogRes, len(vmtrace)),
+	}
+
+	for index, trace := range vmtrace {
+
+		res.StructLogs[index] = StructLogRes{
+			Pc:      trace.Pc,
+			Op:      trace.Op.String(),
+			Gas:     trace.Gas,
+			GasCost: trace.GasCost,
+			Error:   trace.Err,
+			Stack:   make([]string, len(trace.Stack)),
+			Memory:  make(map[string]string),
+			Storage: make(map[string]string),
+		}
+
+		for stackIndex, stackValue := range trace.Stack {
+			res.StructLogs[index].Stack[stackIndex] = fmt.Sprintf("%x", common.LeftPadBytes(stackValue.Bytes(), 32))
+		}
+
+		const maxMem = 10
```

Comment:
This was directly taken from https://github.com/ethereum/go-ethereum/blob/e5532154a50114d5ffb1ffd850b746cab00cb899/core/vm/logger.go#L43

But I guess it would make sense to make this a configurable parameter to be supplied by the user.